rio_tiler/utils.py

```diff
@@ -253,27 +253,33 @@ def non_alpha_indexes(src_dst: Union[DatasetReader, DatasetWriter, WarpedVRT]) -
 
 def linear_rescale(
     image: numpy.ndarray,
-    in_range: Tuple[Union[int, float], Union[int, float]] = (0, 1),
-    out_range: Tuple[Union[int, float], Union[int, float]] = (1, 255),
+    in_range: Tuple[NumType, NumType],
+    out_range: Optional[Tuple[NumType, NumType]] = None,
 ) -> numpy.ndarray:
     """
     Linear rescaling.
 
     Attributes
     ----------
-        image : numpy ndarray
-            Image array to rescale.
-        in_range : list, int, optional, (default: [0,1])
-            Image min/max value to rescale.
-        out_range : list, int, optional, (default: [1,255])
-            output min/max bounds to rescale to.
+    image: numpy ndarray
+        Image array to rescale.
+    in_range: Tuple, int
+        Image min/max value to rescale.
+    out_range: Tuple, optional, (default: (0,255))
+        output min/max bounds to rescale to.
 
     Returns
     -------
-        out : numpy ndarray
-            returns rescaled image array.
+    out: numpy ndarray
+        returns rescaled image array.
 
     """
+    if out_range is None:
+        warnings.warn(
+            "No output range set, using image (0, 255)", UserWarning,
+        )
+        out_range = (0, 255)
```

Comment:
Up to you whether you want to raise this warning. I don't have a preference if the default behavior is just to rescale to 0-255. Makes sense for most use cases.

Reply:
because we don't add `out_range=(0, 255)` in the function definition I think it's important to add this warning.

Reply:
Yeah I'm saying that I don't have a preference between how it is now and removing the warning and including `out_range=(0, 255)` as the default value in the def